src/App.tsx

```diff
@@ -14,49 +15,109 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  CONST_STATE_BY_LENGTH = 'byLength',
+  CONST_STATE_BY_ALPHABET = 'byAlphabet',
+}
+
+function prepareTodos(
+  data: string[],
+  sortBy: SortType | null,
+  isReversed:boolean,
+) {
+  let todos : string[] = [...data];
+
+  if (sortBy === SortType.CONST_STATE_BY_LENGTH) {
+    todos.sort((a, b) => a.length - b.length);
+  } else if (sortBy === SortType.CONST_STATE_BY_ALPHABET) {
+    todos.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (isReversed) {
+    todos = todos.reverse();
+  }
+
+  return todos;
+}
 
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType | null>(null);
+  const [sortStateReverse, setSortStateReverse] = useState(false);
```

Comment:
sortStateReverse it sounds as function, but it's a boolean variable 
so, it's better to rename it to the `isReversed`